code/__HELPERS/game.dm

```diff
@@ -127,29 +127,6 @@
 			turfs += T
 	return turfs
 
-
-//This is the new version of recursive_mob_check, used for say().
-//The other proc was left intact because morgue trays use it.
-/proc/recursive_hear_check(atom/O)
-	var/list/processing_list = list(O)
-	var/list/processed_list = list()
-	var/found_atoms = list()
-
-	while (processing_list.len)
-		var/atom/A = processing_list[1]
-
-		if (A.flags & HEAR)
-			found_atoms |= A
-
-		for (var/atom/B in A)
-			if (!processed_list[B])
```

Comment:
```
if(!processed_list[B])
    processing_list |= B
```

"if it's not in the list, check that it's not in the list stupidly more expensive and add it if it's not"
Honk.